test/channel.js

```diff
@@ -202,8 +202,42 @@ test('event channel', assert => {
 
   assert.ok(unsubscribeErr, 'eventChannel should throw if subscriber does not return a function to unsubscribe')
 
+  let unsubscribeWasCalled = false;
+  let chan = eventChannel(() => () => {
+    unsubscribeWasCalled = true;
+  });
+  chan.close();
+  assert.equal(unsubscribeWasCalled, true, 'eventChannel should call unsubscribe when channel is closed')
+
+  unsubscribeWasCalled = false;
+  chan = eventChannel((emitter) => {
+    emitter(END);
+    return () => {
+      unsubscribeWasCalled = true;
+    }
+  });
+  assert.equal(unsubscribeWasCalled, true, 'eventChannel should call unsubscribe when END event is emitted');
+
+  unsubscribeWasCalled = false;
+  let milliseconds = 2;
+  chan = eventChannel((emitter) => {
+    const interval = setInterval(() => {
+      milliseconds -= 1;
+      if (milliseconds > 0) {
+        emitter(milliseconds);
+      } else {
+        emitter(END);
+      }
+    }, 1);
+    return () => {
+      clearInterval(interval);
+      unsubscribeWasCalled = true;
+    };
+  });
+  setTimeout(() => assert.equal(unsubscribeWasCalled, true, 'complex eventChannel should call unsubscribe when END event is emitted'), 5);
```

Comment:
probably this wont even be checked as this test is synchronously going further till `assert.end()`, I would recomment splitting this whole test in some smaller ones and then in more scoped test call `assert.equal` (like you did) + `assert.end()`